### DISCUSSION TITLE
Right modifiers

### DISCUSSION THREAD

**Discussion by ggppjj** [manna-harbour/miryoku#262] [Q&A] [answered]
Hello,

I am currently modifying the Miryoku layout for my own personal use with a Cheapino split ortho board, and noticed that the default keymapping includes no right modifiers at all. 

I've [updated my own copy](https://github.com/ggppjj/qmk_firmware/blob/365b3f6f351740a3307b4e7c0dd8e99b427117f9/users/manna-harbour_miryoku/miryoku_babel/miryoku_layer_alternatives.h#L89C1-L89C1) to have the right modifiers for the right side of the keyboard, and wanted to raise an issue here for adoption by default if found appropriate.

There are programs that differentiate between left and right control, for example VirtualBox uses the Right control key very specifically to signal that the user wants to break out of the guest window. Using the default keymap and default settings in VirtualBox, I would be trapped in the VM until I switched to another keyboard or found some other silly way around it.

I did a search for "right control" before submitting this.

**Comment by samuelblowes** [comment]
I wondered this, too. Since QMK (not sure about the others) allows to specify _which side_ modifier keys belong to, I was curious as to why the HRMs don't correspond by default.

**Comment by manna-harbour** [comment, marked as answer]
See https://github.com/manna-harbour/miryoku/discussions/221.

**Reply by ggppjj** [reply]
I understand the decision now, and have decided that this layout is unfortunately not suitable for my everyday work. Thank you for your time.

**Reply by manna-harbour** [reply]
To expand on #221, if you need right mods, the solution would be to add them as mod-tap on the bottom row, as is done already for AltGr.

The Control and Shift positions are used for layer lock, but only on double tap on layers, and not at all on Base. They can be added as mod-tap on Base, and as the single tap and the hold functions on the layers.

The Gui position is used for Button, so if Right Gui is required it would replace the layer-tap on Base with a mod-tap, and add a basic mod on the layers, as is done for AltGr.

Miryoku aims to make all the important functionality of a standard keyboard available with maximum possible ergonomics, and make possible all the remaining functionality where practical. App behaviour such as distinguishing left/right mods or taps of mods is arguably broken, and should be handled by rebinding, but will also be accommodated where practical.

Taps of mods are already supported on the layers, and hold or tap of right mods would be supported by the above changes. Out of those, only replacing Button with Right Gui is a breaking change, and could be provided via a configuration option.

**Reply by manna-harbour** [reply]
See also https://github.com/manna-harbour/miryoku/issues/33. This issue could be accommodated as part of making mods customisable in general, and supporting bottom row mods.
